src/cache/messages_intf.ml

```diff
@@ -31,6 +31,7 @@ type incoming = Incoming
 
 (** Messages of the communication protocol between Dune and the cache daemon. *)
 type _ message =
+  | Hint : Digest.t list -> outgoing message
```

Comment:
Please add a brief explanation for the message. I'd also place it closer to `Promote` in the list, after various initialisation messages.